cord.ls

```diff
@@ -4,27 +4,42 @@ require! {
   "./auth"
 }
 
+# Plugins to load from the local folder
 plugins =
   "./log"
   "./relay"
   "./commands"
 
-irc = new IrcServer auth.irc
-discord = new DiscordClient!
-
-for plugin in plugins
-  (require plugin) irc, discord
-
-irc.connect!
-discord.login ...auth.discord<[email password]>
-
-# Quit nicely when pressing CTRL-C
-process.on \SIGINT, !->
+# Finalize
+finalize = !->
   Promise.all do
-    irc.disconnect!
+    irc?.disconnect!
     discord.logout!
   .then do
     # I honestly don't remember if this is an
     # error or actually supposed to be like this.
     !-> process.exit!
     !-> process.exit!
+
+# Connecting to IRC
+if auth.irc?
+  irc = new IrcServer auth.irc
+  console.log "[= IRC =] Connecting to #{auth.irc.server}"
+  irc.connect (err) !-> if err?
+    console.error "[= IRC =] #err"
+    finalize!
+else console.log "[= IRC =] No channel selected!"
+
+discord = new DiscordClient!
+
+# Loading modules, irc parameter is optional, may be undefined
+for plugin in plugins
+  (require plugin) discord, irc
+
+# Connect to discord
+discord.login ...auth.discord<[email password]>, (err) !-> if err?
```

Comment:
Since discord-js supports Promises, might be cleaner to use those instead of callbacks.